packages/babel-plugin-transform-classes/test/fixtures/spec/this-in-params-of-constructor/output.js

```diff
@@ -0,0 +1,25 @@
+var Root = /*#__PURE__*/babelHelpers.createClass(function Root() {
+  "use strict";
+
+  babelHelpers.classCallCheck(this, Root);
+});
+var Foo = /*#__PURE__*/function (_Root) {
+  "use strict";
+
+  var _Foo;
+  (function () {
+    var _this;
+    _Foo = function Foo(cb = () => {
+      return babelHelpers.assertThisInitialized(_this);
+    }) {
+      babelHelpers.classCallCheck(this, _Foo);
+      _this = _super.call(this);
+      _this.cb = cb;
+      return _this;
+    };
```

Comment:
I believe this has problems due to `_this` being in a scope too high: do these assertions pass?

```js
const foo1 = new Foo();
const foo2 = new Foo();

expect(foo1.cb()).toBe(foo1);
expect(foo2.cb()).toBe(foo2);
```

Reply:
Yes, you're right, this fails a test.